File: Assets/Code/Outfit7/Raven/TrackComponents/Properties/AnimationData/RavenAnimationDataPropertyVector3.cc
#ifdef STARLITE
#include "RavenAnimationDataPropertyVector3.h"
#include "RavenAnimationDataPropertyVector3.cs"

namespace Starlite {
    namespace Raven {
        RavenAnimationDataPropertyVector3::RavenAnimationDataPropertyVector3() {
        }

        void RavenAnimationDataPropertyVector3::ProcessValueComponents(Vector3& value, Object* targetComponent) {
            bool synced = false;
            Vector3 syncedValue = Vector3::Zero;

            for (int i = 0; i < c_ValueCount; ++i) {
                // a choice between call + get vs if, I choose if
                if (!m_ApplyValues[i]) {
                    if (!synced) {
                        synced = true;
                        syncedValue = GetValue(targetComponent);
                    }

                    value.data[i] = syncedValue.data[i];
                }
            }
        }
    } // namespace Raven
} // namespace Starlite
#endif

File: Assets/Code/Outfit7/Raven/TrackComponents/Properties/AnimationData/RavenAnimationDataPropertyVector3.cs
//{ "engine_register_objects":["Starlite__Raven__RavenAnimationDataPropertyVector3"] }
#if !__cplusplus
#pragma warning disable
namespace Starlite {

    namespace Raven {
    
        // RavenAnimationDataPropertyVector3
        [Starlite.ObjectClassAttribute("Starlite::Raven::RavenAnimationDataPropertyVector3")]
        public sealed partial class RavenAnimationDataPropertyVector3 : global::Starlite.Raven.RavenAnimationDataPropertyBase<UnityEngine.Vector3> {
        	// Properties
        	[UnityEngine.SerializeField] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] private bool[] m_ApplyValues = new bool[0];
        }
        
    }
    
}
#pragma warning restore
#else
// RavenAnimationDataPropertyVector3
namespace Starlite { namespace Raven { 
SIMPLEMENT_CLASS_WITH_PROPERTIES_IN_CLASS_SCOPED(Starlite::Raven::RavenAnimationDataPropertyVector3, RavenAnimationDataPropertyVector3, RavenAnimationDataPropertyVector3, Starlite::Raven::RavenAnimationPropertyComponentBase, Starlite::SceneObjectComponent, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataPropertyVector3, m_PropertyType, m_PropertyType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataPropertyVector3, m_Offset, m_Offset, kVector3, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataPropertyVector3, m_Multiplier, m_Multiplier, kDouble, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataPropertyVector3, m_ApplyMultiplier, m_ApplyMultiplier, kBool, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataPropertyVector3, m_ApplyOffset, m_ApplyOffset, kBool, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataPropertyVector3, m_ExecuteFunctionOnly, m_ExecuteFunctionOnly, kBool, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataPropertyVector3, m_ApplyValues, m_ApplyValues, kArray, kBool, -1, "")
SIMPLEMENT_CLASS_END(RavenAnimationDataPropertyVector3)
} } 
SIMPLEMENT_CLASS_SCOPE(Starlite__Raven__RavenAnimationDataPropertyVector3, Starlite::Raven, RavenAnimationDataPropertyVector3)
#endif
